internal/cmp/router_firewall_rule_group.go

```diff
@@ -0,0 +1,103 @@
+// (C) Copyright 2021 Hewlett Packard Enterprise Development LP
+
+package cmp
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/client"
+	"github.com/HewlettPackard/hpegl-vmaas-cmp-go-sdk/pkg/models"
+	"github.com/HewlettPackard/hpegl-vmaas-terraform-resources/internal/utils"
+	"github.com/tshihad/tftags"
+)
+
+type routerFirewallRuleGroup struct {
+	routerFirewallRuleGroupClient *client.RouterAPIService
+}
+
+func newRouterFirewallRuleGroup(routerFirewallRuleGroupClient *client.RouterAPIService) *routerFirewallRuleGroup {
+	return &routerFirewallRuleGroup{
+		routerFirewallRuleGroupClient: routerFirewallRuleGroupClient,
+	}
+}
+
+func (r *routerFirewallRuleGroup) Read(ctx context.Context, d *utils.Data, meta interface{}) error {
+	var tfFirewallRuleGroup models.CreateRouterFirewallRuleGroup
+	if err := tftags.Get(d, &tfFirewallRuleGroup); err != nil {
+		return err
+	}
+	// Get the router, if the router not exists, return warning
+	router, err := r.routerFirewallRuleGroupClient.GetSpecificRouter(ctx, tfFirewallRuleGroup.RouterID)
+	if err != nil {
+		return err
+	}
+	// if router not found set is_deprecated flag=true
+	if router.ID == 0 {
+		log.Printf("[ERROR] Router with %d id is not found on Firewall Rule Group plan", tfFirewallRuleGroup.RouterID)
+		tfFirewallRuleGroup.IsDeprecated = true
+
+		return tftags.Set(d, tfFirewallRuleGroup)
+	}
+
+	_, err = r.routerFirewallRuleGroupClient.GetSpecificRouterFirewallRuleGroup(ctx, tfFirewallRuleGroup.RouterID,
+		tfFirewallRuleGroup.ID)
```

Comment:
not capturing response is fine only if we don't have any computed field from the response (other than `id`). I think here this approch is fine